### PR TITLE
Fix NATS high cpu usage

### PR DESCRIPTION
**Pull request by vahpetr**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed high in CPU consumption when working with NATS.

### Documentation entry for user-facing changes
Before, when subscribing to the NATS queue, clickhouse consumed 1 processor core, in the NATS connection and reconnect scenarios, even more CPU consumption occurred, which negatively affected performance in general. With this fix, you will be able to use NATS queues without experiencing unnecessarily high CPU utilization.

### CODE REVIEW

**Comment by vahpetr** [comment]
Сделано для решения проблемы https://github.com/ClickHouse/ClickHouse/issues/41432 https://github.com/ClickHouse/ClickHouse/issues/54353

**Comment by vahpetr** [comment]
This is my first C++ in the last 15 years, so if i have make a mistake, don't scold me :)

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 9c61ceb05100c968ca9665565e91e1007f66802a with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54399/9c61ceb05100c968ca9665565e91e1007f66802a/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by alexey-milovidov — APPROVED** [review]
Thank you! I didn't check all the details of the surrounding code, but it looks alright.

**Comment by evillique** [comment]
After these changes, it is impossible to drop a table with NATS engine. The test that would have found the issue (test_storage_nats) was disabled a long time ago and recently deleted.

File: src/Storages/NATS/NATSHandler.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ void NATSHandler::startLoop()
 
     while (loop_state.load() == Loop::RUN && duration.count() < MAX_THREAD_WORK_DURATION_MS)
     {
-        uv_run(loop, UV_RUN_NOWAIT);
+        uv_run(loop, UV_RUN_DEFAULT);
```
Comment (evillique) [inline src/Storages/NATS/NATSHandler.cpp:46]: Because of this line, to be precise

Comment (vahpetr) [reply src/Storages/NATS/NATSHandler.cpp:46]: Sorry. It's sad, any ideas on how to fix this?

**Pull request merged**